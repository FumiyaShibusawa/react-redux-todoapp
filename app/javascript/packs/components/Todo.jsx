import * as React from "react"

class Todo extends React.Component {
  render(){
    let input;
    return (
      <div className="todo_component">
        { (() => {
          if (this.props.todo_list) {
            return(
              <div>
                <h2>{ this.props.todo_list.name }</h2>
                <ul>
                  { (() => {
                    if (this.props.todo_list.todos) {
                      this.props.todo_list.todos.map( (todo, i) => (
                        <li key={ `${todo.name}_${i}` }>
                        <span>{ todo.status }</span>
                        <span>{ todo.name }</span>
                        </li>
                      ))
                    }
                  })() }
                </ul>
                <div onClick={ this.props.toggleTodoForm } >
                  <i className="fa fa-plus"></i>
                  <span>Todoを追加する</span>
                </div>
                <div id="todo_form" style={{ display: 'none' }}>
                  <input type="text" ref= { node => { input = node } }/>
                  <input type="submit" value="送信" onClick={ () => {
                    if (input) {
                      this.props.addNewTodo(input.value, this.props.num);
                      input.value = ""; // テキストボックス内の値をクリア
                    }
                  } }/>
                </div>
              </div>
            )
          }
        })() }
      </div>
    )
  }
}

export default Todo
