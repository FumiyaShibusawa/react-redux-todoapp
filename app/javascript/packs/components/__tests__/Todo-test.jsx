import * as React from 'react';
import Todo from '../Todo';
import { Form } from '../Todo';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';

const todo_lists = [
  {
    _id: { $oid: "5b2d02e7f450fe5251b9eb9d" },
    user_id: { $oid: "5b2cf7fef450fe50d9d60871" },
    created_at: "2018-06-22T14:08:39.983Z",
    updated_at: "2018-06-22T14:08:39.983Z",
    name: "write code everyday",
    todos: [{
      _id: { $oid: "5b237ade2208a2001bc74882" },
      created_at: "2018-06-22T14:08:39.983Z",
      updated_at: "2018-06-22T14:08:39.983Z",
      name: "open your editor",
      todo_list_id: { $oid: "5b2d02e7f450fe5251b9eb9d" },
      completed: false
    }]
  }
]
describe('<Todo />', () => {
  test('it renders correctly', () => {
    const tree = renderer.create(
      <Todo
        todo_list={todo_lists[0]}
        num={0}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
  // toggleForm
  test('state "isFormToggled" set to true when toggleForm fired', () => {
    const todo = mount(<Todo todo_list={todo_lists[0]} num={0} />);
    todo.find('.add-button-text').simulate('click');
    expect(todo.state('isFormToggled')).toBe(true);
  });

  test('Form Component appears when toggleForm fired', () => {
    const todo = mount(<Todo todo_list={todo_lists[0]} num={0} />);
    todo.find('.add-button-text').simulate('click');
    expect(todo.containsMatchingElement(<Form />)).toBe(true);
  });

  // showTodoMenu
  test('todo-menu appears when showTodoMenu fired', () => {
    const todo = mount(<Todo todo_list={todo_lists[0]} num={0} />);
    todo.find('.menu-ellipsis').simulate('click');
    expect(todo.find('.todo-menu')).toHaveLength(1);
  });
});
