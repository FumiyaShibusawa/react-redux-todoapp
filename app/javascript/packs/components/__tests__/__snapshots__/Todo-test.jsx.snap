// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todo /> it renders correctly 1`] = `
<div
  className="todo_component"
>
  <div>
    <h2>
      write code everyday
    </h2>
    <ul
      className="todos"
    >
      <li>
        <span
          className="todo-element"
        >
          <span
            className="completed false"
            onClick={[Function]}
          />
          open your editor
        </span>
        <span
          className="menu-ellipsis"
          onClick={[Function]}
        >
          ï¸™
        </span>
      </li>
    </ul>
    <div
      className="add-button"
    >
      <div
        data-add="show-todo"
      >
        <div
          className="plus"
        >
          +
        </div>
        <span
          className="add-button-text"
          onClick={[Function]}
        >
          add new todo
        </span>
      </div>
    </div>
  </div>
</div>
`;
